### ISSUE TITLE
On create page, add default locale automatically

### ISSUE THREAD

**Issue by sanderha** [tractorcow/silverstripe-fluent#402]
This issue is for when `FluentFilteredExtension` is enabled on `SiteTree`.

When creating a new page, it has no Locales by default. So even after publishing the page, it will not show up in frontend, unless a Locale is added.
Wouldnt it make sense to automatically add the global default domain to a newly created page? Then the user can always change it if they like.

I added this through an extension:
```php
    public function onBeforeWrite()
    {
        // add default locale to newly create page automatically
        if (!$this->owner->isInDB() && !$this->owner->FilteredLocales()->exists()) {
            $default = DataObject::get_one(Locale::class, ["IsGlobalDefault" => 1]);
            $this->owner->FilteredLocales()->add($default);
        }
    }
```

**Comment by robbieaverill** [comment]
@sanderha can you please provide the versions of SilverStripe and the fluent module you're using?

**Comment by robbieaverill** [comment]
Yeah this issue is somewhat related to https://github.com/tractorcow/silverstripe-fluent/issues/350 which we treated as a documentation issue.

This idea might be worth exploring though. What do you think @tractorcow?

**Comment by tractorcow** [comment]
The default locale used to mean a lot more in fluent 3.x than it does in 4.x. The new convention is that all locales are equal, and that there is no real concept of a "default". The only reason for keeping the default locale a flag is to assist with choosing the view locale in the frontend. I don't think that "assume this page is always visible in the default locale" is a good default to be honest.

**Comment by sanderha** [comment]
@tractorcow I think this issue is mainly about `FluentFilteredExtension`. Without it, everything works fine. But if its on, every page is disabled by default, because they have no locales attached. IMO in this case its best to attach default locale on create so clients dont get confused

**Comment by sanderha** [comment]
@tractorcow I dont think `FluentFilteredExtension` should be on by default. I just think that *if* its on, the global default locale should be attached to all pages. Then clients can remove it if they want

**Comment by tractorcow** [comment]
Sorry, I mis-spoke.

I meant "should it be enabled for all locales by default?". That may be less destructive than requiring users to re-check locales.

**Comment by sanderha** [comment]
Hmm yeah it probably should just enable all locales when creating a page. Thats also how Fluent 3 worked I believe (all locales were checked).

**Comment by tractorcow** [comment]
I think DataObject needs an onCreated extension point which is invoked only once for every model. then we could add the locale links there.

**Comment by RoyalPulp** [comment]
Is there any progress on this? 
For large projects it is a catastrophe to add this FluentFilterExtension to SiteTree  later to live running Websites.
 -> When you have many pages in all languages and your customer asks: "Is there a possibility to hide this page just in one language" - the moment you add the extension for SiteTree to the config, the whole website is "offline"/ 404 in all locales.
With SS3.x there has not been a problem with that.
All locales should be active by default! 

This should better be a "hide locale" feature than a "add locale"

**Comment by sanderha** [comment]
@RoyalPulp We've made a task that attaches locales to pages after the FluentFilteredExtension is added, which might help you out: 
https://github.com/nobrainer-web/silverstripe-fluent-recipe/blob/master/src/BuildTasks/ToggleFilteredLocales.php

**Comment by tractorcow** [comment]
I might pick this up; Thanks @sanderha for that I'll look through your recipe later. :) It has some good ideas.
